Makefile.include

```diff
@@ -14,10 +14,11 @@ RIOTPKG        ?= $(RIOTBASE)/pkg
 RIOTTOOLS      ?= $(RIOTBASE)/dist/tools
 RIOTPROJECT    ?= $(shell git rev-parse --show-toplevel 2>/dev/null || pwd)
 GITCACHE       ?= $(RIOTTOOLS)/git/git-cache
+BUILDOUT_DIR   ?= $(RIOTBASE)/build
 APPDIR         ?= $(CURDIR)
-BINDIRBASE     ?= $(APPDIR)/bin
-BINDIR         ?= $(BINDIRBASE)/$(BOARD)
-PKGDIRBASE     ?= $(BINDIRBASE)/pkg/$(BOARD)
+BINDIRBASE     ?= $(BUILDOUT_DIR)/bin
```

Comment:
As explained in the main comments, I think the ones currently in `bin` should not been moved for the moment.
It is an API change I would not merge in the last weeks of a release.

Adding a new directory that needs to start being handled for new commands is ok.
The internal organization of this directory will also need some thinking.

Also, there is still some work to make this correctly handled by `docker` if it is set outside of RIOT. This could help https://github.com/RIOT-OS/RIOT/pull/9646
No idea what the `scan-build` does.